src/sqllogictest/runner.rs

```diff
@@ -490,7 +490,9 @@ impl State {
             }
             Err(error) => {
                 if let Some(expected_error) = expected_error {
-                    if Regex::new(expected_error)?.is_match(&error.to_string()) {
+                    if Regex::new(expected_error)?.is_match(&error.to_string())
+                        || expected_error == error.to_string()
```

Comment:
Unfortunately this change will make our SLT runner incompatible with Cockroach test files. Instead you'll need to escape the regex metacharacters in the test files.